compatibility.json

```diff
@@ -3,6 +3,8 @@
   "minOSVersion": {
     "darwin": "10.15",
     "linux": "2.6.32",
-    "windows": "10.0"
+    "windows": "10.0",
+    "darwin@go1.23": "11",
+    "linux@go1.24": "3.10"
```

Comment:
What's this for?

Reply:
It's just documentation, so the dev who needs to update this file, won't have to do any research, but simply cut-n-paste.

Reply:
Mmm let's not do that, it's confusing and how do you know what will be the requirements for Go 1.24? This should be updated at release time, or at least just shortly before.

What we could do to be more flexible is to have a source file with multiple sets of data in it, for example in yaml format:

```
- runtime: go1.21
   requirements:
     darwin: 20
     windows: 10.0
     ...
- runtime: go1.22
   requirements:
     darwin: 21
     windows: 10.1
     ...
```

and then we could at build time pick the set based on runtime.Version, and then it could be a build failure (for everyone) when there aren't any requirements for the current version. Then you could in principle add things for the future as well, though I think we should be careful about that.

Reply:
@calmh I implemented your suggestion, so compatibility.json is now generated dynamically from [compatibility.yaml](https://github.com/rasa/syncthing/blob/feature/9290-disable-upgrades/compatibility.yaml) based on the current `runtime.Version()`.